pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1091,6 +1091,9 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
                     }
                 });
             }
+            final CompletableFuture<Optional<Topic>> topicFutureToRemove = topicFuture;
+            topicFuture.whenComplete((ignore, ex) -> topics.remove(topicName.toString(), topicFutureToRemove));
```

Comment:
please note that you are appending a new whenComplete listener every time this method is called and the object is already in the map.
we have to call "whenComplete" only if the body of  the lambda expression in computeIfAbsent is executed.
I know that it sounds ugly, but we must to it

Reply:
@eolivelli 

Ah, sorry. I lost a check: `if (ex != null)`. It only is removed if the future was failed, and we call `remove(k, v)`, and this method will not do incorrect deletions. Could you take a look again?

Reply:
The problem is that we call whenComplete every time we invoke this method and only the first time we create the object and add it to the Map.

So everytime we enqueue the execution. If this method is called in an hotpath you will see many allocations

Reply:
@eolivelli 

You are right. Thanks, already fixed